### ISSUE TITLE
Incorrect highlighting for multidimensional slice type in struct's fields

### ISSUE THREAD

**Issue by Vacym** [worlpaker/go-syntax#3]
Hello! I noticed that 2d-slices are not highlighting, when it is field of structure. for example:
```go
package main

type cell struct{}

type board struct {
	matrixCells        [][]cell
	matrixCellsPointer *[][]cell
	rowCells           []cell
	rowCellsPointer    *[]cell

	matrixInts        [][]int
	matrixIntsPointer *[][]int
}

func main() {
	test := new(board)
}
```
![image](https://user-images.githubusercontent.com/65008219/231712135-9d34bf23-b924-4131-a9db-2430ba904800.png)
_Monokai theme_

Interestingly, if you assign a pointer to a 2d slice, the highlighting works

This problem also appears with as many-dimensional sliсes as with arrays.

**Comment by worlpaker** [comment]
Hi,

Thank you for your time and contribution.

If you weren't assigned a pointer before slices, the highlighting wasn't working.

**v0.2.9 - Dark+ Theme**
![fixed_3](https://user-images.githubusercontent.com/34960716/231760633-106982bc-1438-4d8b-a88c-68016f155b6c.png)

---

This bug has been fixed in v0.2.9 with your help, thanks!

Please let me know of any incorrect highlights after the version.

**Comment by Vacym** [comment]
Yes, I didn't notice any problem. Thanks!
